source/fundamentals/crud/write-operations/change-a-document.txt

```diff
@@ -0,0 +1,203 @@
+
+Update operations modify the fields that you specify in one or more
+documents while leaving other fields and values unchanged. Replace
+operations remove all existing fields except for ``_id`` in
+one or more documents and substitute the deleted fields with new fields
+and values.
+
+A Note About ``_id``
+~~~~~~~~~~~~~~~~~~~~
+
+Each document in a MongoDB collection has a unique and immutable ``_id``
+field. You cannot use update and replace operations to modify the
+``_id`` values. If you attempt to change this field, the update and
+replace methods return a ``WriteError``.
+
+.. _updateDocuments:
+
+Update
+------
+
+To update one document, use the ``UpdateOne()`` method, and to update
+multiple documents, use the ``UpdateMany()`` method. Each method takes
+an **update document** that includes at least one **update operator**.
+The update operator specifies the type of update to perform. The update
+document also includes the fields and values that describe the change.
+Update documents use the following format:
+
+.. code-block:: go
+
+   bson.D{{"<update operator>", bson.D{{"<field>", "<value>"}, 
+                                       {"<field>", "<value>"}, ... }}, 
+          {"<update operator>", ... }, ... }
+
+See the MongoDB server manual for a :manual:`complete list of update operators
+and descriptions </reference/operator/update-field/>`.
+
+.. note:: Aggregation Pipelines in Update Operations
+
+   If you are using MongoDB Version 4.2 or later, you can use aggregation
+   pipelines made up of a subset of aggregation stages in update operations. For
+   more information on the aggregation stages MongoDB supports in
+   aggregation pipelines, see our tutorial on performing
+   :manual:`updates with aggregation pipelines </tutorial/update-documents-with-aggregation-pipeline/>`.
+
+Example
+~~~~~~~
+
+The following document describes an employee:
+
+.. code-block:: json
+   :copyable: false
+
+   {
+      "_id" : 2158,
+      "name" : "Mary Shelley",
+      "department" : "Marketing",
+      "role" : "Marketing Analyst",
+      "bonus" : 2500,
+      ...
+   }
+
+The following code shows how you can use ``UpdateOne()`` to:
+
+- Locate the document with ``_id`` value 2158.
+- Use the ``$set`` update operator to change the value of the ``name``
+  field to "Mary Wollstonecraft Shelley" and the ``role`` field to
+  "Marketing Director".
+- Use the ``$inc`` update operator to increment the value of the
+  ``bonus`` field by 2000.
+
+.. code-block:: go
+
+   filter := bson.D{{"_id", 2158}}
+   update := bson.D{{"$set", bson.D{{"name", "Mary Wollstonecraft Shelley"}, 
+       {"role", "Marketing Director"}}, {"$inc", bson.D{{"bonus", 2000}}}}
+
+   result, err := collection.UpdateOne(context.TODO(), filter, update)
+
+``UpdateOne()`` and ``UpdateMany`` return an ``UpdateResult`` type that
+contains information about the update operation such as
+``MatchedCount``, ``ModifiedCount``, ``UpsertedCount``, ``UpsertedID``.
```

Comment:
Suggestion: I think this should be moved up to before the example section. And instead of listing exactly what the calls are, you can change it to a bulleted list explaining what each information is.

i.e. "The UpdateResult type provides the following information: "

Reply:
addressed